### PR TITLE
In 148607219 sibling at risk validation

### PR DESCRIPTION
**Pull request by jgedeon-osi**

### CODE REVIEW

File: spec/screening/sibling_at_risk_spec.rb [hunk 1]
```diff
@@ -170,9 +226,17 @@
           end
 
           screening_page.set_allegations_attributes(allegations: [
-            {victim_id: victim1[:id], perpetrator_id: perpetrator1[:id], allegation_types: ['General neglect']},
-            {victim_id: victim2[:id], perpetrator_id: perpetrator2[:id], allegation_types: ['Physical abuse']}
-          ])
+                                                      {
```
Comment (NateOSI) [inline spec/screening/sibling_at_risk_spec.rb]: I think we might want to use a builder pattern going forward, i.e.
```
ScreeningPage.add_allegation()
.with_victim_id(victim_id)
.and_perpetrator_id(perpetrator_id)
.and_maltreatments(['Neglect']).save!
```

Comment (jgedeon-osi) [inline spec/screening/sibling_at_risk_spec.rb]: I think this is a good idea, I'll work it in as part of a different chore refactor.

File: spec/screening/sibling_at_risk_spec.rb [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+          within '#allegations-card' do
+            expect(page).to have_content(error_message)
+            click_link 'Edit'
+            expect(page).to have_content(error_message)
+          end
+        end
+      end
+
+      context 'has added other allegations to a different perpetrator' do
```
Comment (ThomasOSI) [inline spec/screening/sibling_at_risk_spec.rb]: recommend rephrasing as 'has only added ...'

File: spec/screening/sibling_at_risk_spec.rb [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+          within '#allegations-card' do
+            expect(page).not_to have_content(error_message)
+            click_link 'Edit'
+            expect(page).not_to have_content(error_message)
+          end
+        end
+      end
+
+      context 'has added other allegations to a different perpetrator' do
```
Comment (ThomasOSI) [inline spec/screening/sibling_at_risk_spec.rb]: ... only ...

File: spec/screening/sibling_at_risk_spec.rb [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+          ])
+
+          within '#allegations-card' do
+            expect(page).to have_content(error_message)
+            click_link 'Edit'
+            expect(page).to have_content(error_message)
+          end
+
+          screening_page.set_allegations_attributes(allegations: [
+            {victim_id: victim1[:id], perpetrator_id: perpetrator1[:id], allegation_types: ['At risk, sibling abused']},
```
Comment (ThomasOSI) [inline spec/screening/sibling_at_risk_spec.rb]: Isn't this already set?  If so, recommend removing the line so it's more clear that setting General neglect is what changes the expectations.

File: spec/screening/sibling_at_risk_spec.rb [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+            expect(page).not_to have_content(error_message)
+            click_link 'Edit'
+            expect(page).not_to have_content(error_message)
+          end
+        end
+      end
+    end
+  end
+
+  context 'Removing the conditionaal business rule' do
```
Comment (ThomasOSI) [inline spec/screening/sibling_at_risk_spec.rb]: "conditionaal" => "conditional"

File: spec/screening/sibling_at_risk_spec.rb [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+          within '#allegations-card' do
+            expect(page).to have_content(error_message)
+            click_link 'Edit'
+            expect(page).to have_content(error_message)
+          end
+
+          screening_page.set_allegations_attributes(allegations: [
+            {victim_id: victim1[:id], perpetrator_id: perpetrator1[:id], allegation_types: ['At risk, sibling abused']},
+            {victim_id: victim2[:id], perpetrator_id: perpetrator1[:id], allegation_types: ['Physical abuse']}
```
Comment (ThomasOSI) [inline spec/screening/sibling_at_risk_spec.rb]: I don't know which type of fix to expect from the `context` and `it` statements.  Recommend changing the `it` to indicate the nature of the fix. e.g. `it 'is satisfied when complementary allegation added'`

File: spec/screening/sibling_at_risk_spec.rb [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+            expect(page).to have_content(error_message)
+          end
+        end
+      end
+    end
+  end
+
+  context 'Fixing missing fields' do
+    context 'when adding one or more allegations of "At risk, sibling abused"' do
+      context 'has not added any complementary allegation to a different allegation' do
```
Comment (ThomasOSI) [inline spec/screening/sibling_at_risk_spec.rb]: Confused by this context. To a different victim?

File: spec/screening/sibling_at_risk_spec.rb (hunk 2 above)
Comment (jgedeon-osi) [inline spec/screening/sibling_at_risk_spec.rb]: Done.

File: spec/screening/sibling_at_risk_spec.rb (hunk 3 above)
Comment (jgedeon-osi) [inline spec/screening/sibling_at_risk_spec.rb]: Done.

File: spec/screening/sibling_at_risk_spec.rb (hunk 5 above)
Comment (jgedeon-osi) [inline spec/screening/sibling_at_risk_spec.rb]: done.

File: spec/screening/sibling_at_risk_spec.rb (hunk 6 above)
Comment (jgedeon-osi) [inline spec/screening/sibling_at_risk_spec.rb]: Done

File: spec/screening/sibling_at_risk_spec.rb (hunk 7 above)
Comment (jgedeon-osi) [inline spec/screening/sibling_at_risk_spec.rb]: Fixed

**Pull request merged**